src/components/TileCatalog/TileCatalog.test.jsx

```diff
@@ -1,18 +1,145 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render, fireEvent } from '@testing-library/react';
 
 import TileCatalog from './TileCatalog';
-import { commonTileCatalogProps } from './TileCatalog.story';
+import SampleTile from './SampleTile';
+
+const getTiles = num => {
+  const tiles = [];
+  Array(num)
+    .fill(0)
+    .map((i, idx) => {
+      tiles[idx] = (
+        <SampleTile title={`Tile ${idx + 1}`} description="This is a sample product tile" />
+      );
+      return tiles[idx];
+    });
+  return tiles;
+};
 
 describe('TileCatalog tests', () => {
-  test('error prop', () => {
-    // If we have data, never show the error state
-    const wrapper = shallow(<TileCatalog {...commonTileCatalogProps} error="In error state" />);
-    expect(wrapper.find('TileCatalog__StyledEmptyTile')).toHaveLength(0);
-    // If we have empty data then show the error state
-    const wrapper2 = shallow(
-      <TileCatalog {...commonTileCatalogProps} tiles={[]} error="In error state" />
+  test('TileCatalog gets rendered', () => {
+    const { getByText } = render(
+      <TileCatalog
+        tiles={getTiles(2, 'Tile')}
+        title="Test Tile Catalog"
+        numRows={1}
+        numColumns={2}
+      />
+    );
+    expect(getByText('Tile 1')).toBeTruthy();
+    expect(getByText('Tile 2')).toBeTruthy();
+  });
+
+  test('TileCatalog placeholder tile are rendered', () => {
+    const { getByText } = render(
+      <TileCatalog
+        tiles={getTiles(6, 'Tile')}
+        title="Test Tile Catalog"
+        numRows={2}
+        numColumns={2}
+      />
+    );
+    expect(getByText('Tile 1')).toBeTruthy();
+    expect(getByText('Tile 2')).toBeTruthy();
+  });
+
+  test('TileCatalog to have default ', () => {
```

Comment:
This is checked by PropTypes already so a test is redundant

Reply:
It is checked by PropTypes. The jest test coverage still asks me to cover the default function. So I added this test. Will update the name a bit.